numpy/lib/index_tricks.py

```diff
@@ -146,24 +146,25 @@ def __init__(self, sparse=False):
     def __getitem__(self, key):
         try:
             size = []
-            typ = int
+            # Mimick the behavior of `np.arange` and use a data type
```

Comment:
```suggestion
            # Mimic the behavior of `np.arange` and use a data type
```

Reply:
Done as of https://github.com/numpy/numpy/pull/16971/commits/8a5e1d58b8e57819a9d33aa356dbef43f690cfd3.